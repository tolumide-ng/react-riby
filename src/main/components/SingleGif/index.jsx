import React from 'react';
import { withRouter, Link } from 'react-router-dom';

const SingleGif = ({ imgSrc, title, id, history = {} }) => {
  return (
    <Link
      className="w-1/4 border h-64 m-4 cursor-pointer"
      to={`/${id}`}
    >
      <img
        src={imgSrc}
        alt={title}
        id={id}
        className="h-64 w-full object-cover"
      />
    </Link>
  );
};

export default withRouter(SingleGif);
