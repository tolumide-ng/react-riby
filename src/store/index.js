import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './modules';

const middlewares = applyMiddleware(ReduxThunk);

const store = createStore(rootReducer, composeWithDevTools);

export default store;
