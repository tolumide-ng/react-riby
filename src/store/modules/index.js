import { combineReducers } from 'redux';
import allGifReducer from './allGifs/reducers';
import detailedGifReducer from './detailedGif/reducers';
import searchGifReducer from './searchGif/reducers';

const rootReducer = combineReducers({
  allGifReducer,
  detailedGifReducer,
  searchGifReducer,
});

export default rootReducer;
