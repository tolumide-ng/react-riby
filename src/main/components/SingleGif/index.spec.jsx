import { render, fireEvent, cleanup } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import SingleGif from './index';
import store from '../../../store';

// imgSrc, title, id, history = {}
const imgSrc = '';
const title = 'title';
const id = 'longId';
const history = { push: jest.fn() };

describe('Single Component', () => {
  test('Shoul return a single component', () => {
    const defaultProps = { imgSrc, title, id, history };

    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <SingleGif {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );

    expect(asFragment).toMatchSnapshot();
  });
});
