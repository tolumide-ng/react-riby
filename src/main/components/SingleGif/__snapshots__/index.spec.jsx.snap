// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Component Shoul return a single component 1`] = `[Function]`;
